src/plugins/dashboard/public/dashboard_container/embeddable/create/performance/query_performance_tracking.ts

```diff
@@ -20,57 +20,48 @@ import { DashboardLoadType } from '../../../types';
 let isFirstDashboardLoadOfSession = true;
 
 const loadTypesMapping: { [key in DashboardLoadType]: number } = {
-  sessionFirstLoad: 0,
-  dashboardFirstLoad: 1,
-  dashboardSubsequentLoad: 2,
+  sessionFirstLoad: 0, // on first time the SO is loaded
+  dashboardFirstLoad: 1, // on initial load navigating into it
+  dashboardSubsequentLoad: 2, // on filter-refresh
 };
 
-export const startQueryPerformanceTracking = (
+export function startQueryPerformanceTracking(
   dashboard: PresentationContainer & TracksQueryPerformance
-) => {
-  const reportPerformanceMetrics = ({
-    timeToData,
-    panelCount,
-    totalLoadTime,
-    loadType,
-  }: {
-    timeToData: number;
-    panelCount: number;
-    totalLoadTime: number;
-    loadType: DashboardLoadType;
-  }) => {
-    const duration =
-      loadType === 'dashboardSubsequentLoad' ? timeToData : Math.max(timeToData, totalLoadTime);
-
-    reportPerformanceMetricEvent(coreServices.analytics, {
-      eventName: DASHBOARD_LOADED_EVENT,
-      duration,
-      key1: 'time_to_data',
-      value1: timeToData,
-      key2: 'num_of_panels',
-      value2: panelCount,
-      key4: 'load_type',
-      value4: loadTypesMapping[loadType],
-    });
-  };
-
+) {
   return dashboard.children$
     .pipe(
-      switchMap((children) => {
+      skipWhile((children) => {
+        // Don't track render-status when the dashboard is still adding embeddables.
+        return Object.values(children).length !== dashboard.getPanelCount();
+      }),
+      map((children) => {
+        // Filter for embeddables which publish phase events
         const childPhaseEventTrackers: PublishesPhaseEvents[] = [];
-        for (const child of Object.values(children)) {
-          if (apiPublishesPhaseEvents(child)) childPhaseEventTrackers.push(child);
+        const values = Object.values(children);
+        for (const child of values) {
+          if (apiPublishesPhaseEvents(child)) {
+            childPhaseEventTrackers.push(child);
+          }
+        }
+        return childPhaseEventTrackers;
+      }),
+      switchMap((children) => {
+        if (children.length === 0) {
+          return of([]); // map to empty stream
```

Comment:
When modifying this Observable pipeline I had mistakingly omitted this line https://github.com/elastic/kibana/pull/194349/files#diff-4b6cc0dec90b45b66bc2659772bfa29fe23057c3656f47db8a1f042ec7b2a48bL64

added this back in here https://github.com/elastic/kibana/pull/194349/commits/6f75d7bf90947ba0ec82e53f6c11a4c9ba0aa657 

This fixes the failing unit tests.